WombatProject/Services/GoogleBooksAPIService.cs

```diff
@@ -7,58 +8,64 @@
-        public List<Book> BookItems { get; set; } = new List<Book>();
+        protected readonly GoogleApiSettings _googleApiSettings;
+        public List<Book> AuthorBookItems { get; } = new List<Book>();
+        public List<Book> SearchResults { get; } = new List<Book>();
 
-        public GoogleBooksAPIService(IConfiguration configuration, WombatBooksContext context)
+        public GoogleBooksAPIService(IOptions<GoogleApiSettings> settings)
         {
-            _configuration = configuration;
-            Context = context;
+            this._googleApiSettings = settings.Value;
         }
 
-        public async Task GetSearchResults(string searchTerm)
+        public async Task GetSearchResultsAsync(string searchTerm)
         {
-            BookItems.Clear();
-            string url = $"{_configuration["GBooksURL"]}?q={searchTerm}&maxResults=40";
+            SearchResults.Clear();
+            string url = $"{_googleApiSettings.GoogleBooksURL}?q={searchTerm}&maxResults=40";
             IList<JToken> tokens = await GetBookItemsAsJToken(url);
             foreach (JToken token in tokens)
             {
-                BookItems.Add(parseJsonToken(token));
+                SearchResults.Add(ParseJsonToken(token));
             }
         }
 
-        public async Task GetAuthorBooks(string author)
+        public async Task GetAuthorBooksAsync(string author)
         {
             AuthorBookItems.Clear();
-            string url = $"{_configuration["GBooksURL"]}?q=inauthor:{author}";
+            string url = $"{_googleApiSettings.GoogleBooksURL}?q=inauthor:{author}";
             IList<JToken> tokens = await GetBookItemsAsJToken(url);
             foreach (JToken token in tokens)
             {
-                AuthorBookItems.Add(parseJsonToken(token));
+                AuthorBookItems.Add(ParseJsonToken(token));
             }
         }
 
         private async Task<IList<JToken>> GetBookItemsAsJToken(string url)
         {
-            using (var client = new HttpClient())
+            var client = new HttpClient();
+            var uri = new Uri(url);
+            var response = await client.GetAsync(uri);
+            string textResult = await response.Content.ReadAsStringAsync();
+            JObject bookSearch = JObject.Parse(textResult);
+            IList<JToken> tokens = new List<JToken>();
+            try
             {
-                var uri = new Uri(url);
-                var response = await client.GetAsync(uri);
-                string textResult = await response.Content.ReadAsStringAsync();
-                JObject bookSearch = JObject.Parse(textResult);
-                IList<JToken> tokens = bookSearch["items"].Children().ToList();
-                return tokens;
+                tokens = bookSearch["items"].Children().ToList();
             }
+            catch (NullReferenceException error)
+            {
+                Console.WriteLine(error);
```

Comment:
It should be logged via a Logger instance and not directly to console. Please create an issue about it, so you can later implement a proper logging mechanism.

Reply:
This is no longer an issue since I've reworked this part of the project, based on your above comment.